reframe/frontend/executors/__init__.py

```diff
@@ -53,7 +120,7 @@ def _notify_listeners(self, callback_name):
     def _safe_call(self, fn, *args, **kwargs):
         self._current_stage = fn.__name__
         try:
-            with logging.logging_context(self._check) as logger:
+            with logging.logging_context(self.check) as logger:
```

Comment:
I am a bit confused... Shouldn't we pass tests cases instead of checks here?

Reply:
You should not be confused with this. It's exactly as before. The only difference is that the `RegressionTask` now stores a `TestCase` rather than a check. The reason for that is that for supporting retries, we need to generate a new `TestCase` from a failed task. For that, we need a handle to the original check, which now we don't have any more, cos `runall()` acts on test cases and the cloning happens outside. So this information is stored in the `__check_orig` field of the `TestCase`.